### PR TITLE
Add Markdown rendering

### LINKED ISSUE #229: Render comments as markdown

### PR DESCRIPTION
**Pull request by Huy-Ngo**
Supposed to resolve #229 

Draft because it doesn't work yet.

### CODE REVIEW

**Comment by McSinyx** [comment]
Heads up: [PyMdown is abandoned](https://github.com/facelessuser/PyMdown/commit/5658e79e73ef206dfeb446d8637e04da0dfa5647), I'd recommend python-markdown with bleach as a substitution.

**Comment by Huy-Ngo** [comment]
What's bleach?

**Comment by Huy-Ngo** [comment]
Oh it's the name of the lib

**Comment by Huy-Ngo** [comment]
I'm not quite satisfied with this. Shall we probably avoid `<h1>` by adding and extra # for every markdown?

File: src/acanban/__init__.py [hunk 1]
```diff
@@ -82,6 +85,12 @@ async def test_app(self) -> AsyncIterator[Acanban]:
 app.add_template_filter(naturalsize)
 
 
+@app.template_filter('as_markdown')
+def as_markdown(text):
+    html = clean(markdown(text), markdown_tags, markdown_attrs)
+    return html
```
Comment (McSinyx) [inline src/acanban/__init__.py:91]: Why don't you return it directly?  BTW how about naming the *filter* solely *markdown* to make it look more *nix™?

Regarding the h{1..6}, for comments no one ever uses headings apart from meme purposes, and project descriptions (which you forgot) users can edit when they find out the headings are too large.  There's no need to worry, they're all consenting adults.

File: src/acanban/templates/task.html [hunk 2]
```diff
@@ -26,7 +26,7 @@
   {% set id = comment.created_on.timestamp() %}
   <label for=toggle-{{ id }} class=reply-label>reply</label>
   <input type=checkbox class=reply-toggle id=toggle-{{ id }}><br>
-  {{ comment.description if istask else comment.content }}<br>
+  {{ comment.description | as_markdown | safe if istask else comment.content | as_markdown | safe }}<br>
```
Comment (McSinyx) [inline src/acanban/templates/task.html:29]: This looks redundant.

File: src/acanban/__init__.py (hunk 1 above)
Comment (Huy-Ngo) [reply src/acanban/__init__.py:91]: > how about naming the filter solely markdown

That would overwrite with the function used to parse markdown.

**Comment by Huy-Ngo** [comment]
How do I write stubs for bleach-allowlist?

File: src/acanban/__init__.py (hunk 1 above)
Comment (McSinyx) [reply src/acanban/__init__.py:96]: Use `@app.template_filter('markdown')`.

**Comment by McSinyx** [comment]
[Seems like it's `markdown_tags: List[str]` and `markdown_attrs: Dict[str, List[str]]`](https://github.com/yourcelf/bleach-allowlist/blob/05484ab6a6da1ccedee25709595f11f0981b3192/bleach_allowlist/bleach_allowlist.py#L52-L68).

File: src/acanban/templates/task.html [hunk 3]
```diff
@@ -26,7 +26,7 @@
   {% set id = comment.created_on.timestamp() %}
   <label for=toggle-{{ id }} class=reply-label>reply</label>
   <input type=checkbox class=reply-toggle id=toggle-{{ id }}><br>
-  {{ comment.description if istask else comment.content }}<br>
+  {{ (comment.description if istask else comment.content) | markdown | safe }}<br>
```
Comment (McSinyx) [inline src/acanban/templates/task.html:29]: Just checked it out locally, you'll need to drop the `br` here and maybe use `bleach.linkify` as well.  We'll also need to figure out how to refer to tasks but it's probably out of scope.

**Comment by McSinyx** [comment]
BTW bleach-allowlist is composed of only a few static data, you might want to work with the maintainer to get the stubs upstream (later).

**Review by McSinyx — APPROVED** [review]
It's good, I guess.

**Pull request merged**